src/promises.js

```diff
@@ -17,42 +18,67 @@ const WATCHED_PROMISE_IMPLEMENTATIONS = new Map();
  * @returns {Function|*}
  */
 function rebindResolver(fn, localIdx) {
-  if (typeof fn !== 'function') {
+  if (typeof fn !== 'function' || isCleaningUp) {
     return fn;
   }
-  return function reboundResolver() {
-    strayPromises = strayPromises.filter(({ id }) => id !== localIdx);
+  function reboundResolver() {
+    strayPromises.some((strayPromise) => {
+      if (strayPromise.id !== localIdx || isCleaningUp) {
+        return false;
+      }
+      strayPromise.hasBeenCalled = true;
+      return true;
+    });
     return fn.apply(this, arguments);
-  };
+  }
+  // this is useful for debugging
+  // reboundResolver.originalFn = String(fn);
```

Comment:
(Not a blocker) Considering enabling this on `/jasmine-stray-promises/.test(process.env.NODE_DEBUG)` instead (or something similar).